SIS_import/sis_set_url.rb

```diff
@@ -59,31 +58,57 @@
 	"allowed_call_number_during_interval" => 3000
 }
 
+@CurriculumAdminScope="curriculumadmin"
+@CurriculumScope="umscheduleofclasses"
+
+@esb_api_scope_to_token_hash={
+    @CurriculumAdminScope => nil,
+    @CurriculumScope =>nil
+}
+
 # the alert email address is read from the configuration file
 # and defaults to "canvas-sis-data-alerts@umich.edu", if not set otherwise
 @alert_email_address = "canvas-sis-data-alerts@umich.edu"
 
+
 ## refresh token for ESB API call
-def getESBToken
-	if (@esbToken.nil?)
-		## get new ESB token
-		refreshESBToken();
-	end
-	if (@esbToken.nil?)
-		# return empty string instead of nil
-		return "";
-	else
-		return @esbToken
-	end
+def getESBToken(scope,token_renewal_needed)
+  @esb_api_scope_to_token_hash.each do |key, value|
+    @logger.debug "#{key}: #{value}"
+  end
+  if !token_renewal_needed
+    # Token is still good and can be reused for the particular scope api call
+    @esb_api_scope_to_token_hash.each do |esb_token_key, esb_token_for_scope|
+      if esb_token_key === scope
+        if !esb_token_for_scope.nil?
+          @logger.info "ESB token taken from esb_api_scope_to_token_hash ending with  " + esb_token_for_scope[-5,5] + "... for scope "+esb_token_key
```

Comment:
why use @esb_token_for_scope[-5,5] here?

Reply:
What I have seen when testing the token was usually when generating token right away the last digits seems to be different than the starting. So for visual purposes, I said to print the last 5 of the token than first 5 [0..5] as we have 2 token generation happening since of 2 different scopes.